scripts/downstream-tags.py

```diff
@@ -0,0 +1,191 @@
+            tag_name = ref.replace('refs/tags/', '')
+            if version_pattern.match(tag_name):
+                version_tags.append(tag_name)
+
+        if not version_tags:
+            return None
+
+        # Sort version tags
+        def version_key(tag):
+            # Split into parts: v4.17.0-rc1 -> [4, 17, 0, 1]
+            parts = tag[1:].split('.')  # Remove 'v' prefix
+            major, minor = parts[0:2]
+            patch_rc = parts[2].split('-')
+            patch = int(patch_rc[0])
+            # RC versions should sort before final versions
+            rc = int(patch_rc[1][2:]) if len(patch_rc) > 1 else float('inf')
+            return (int(major), int(minor), patch, rc)
+
+        return max(version_tags, key=version_key)
+    except Exception as e:
+        print(f"Error fetching tags for {repo['name']}: {e}", file=sys.stderr)
+        return None
+
+def check_toolchain_history(repo: Dict[str, str], version: str) -> Optional[str]:
+    """Check git history of lean-toolchain file for first occurrence of version"""
+    github_url = repo['github']
+    repo_name = github_url.replace('https://github.com/', '')
+
+    try:
+        result = subprocess.run(
+            ['gh', 'api', f'repos/{repo_name}/commits?path=lean-toolchain'],
+            capture_output=True,
+            text=True
+        )
+        if result.returncode != 0:
+            return None
+
+        commits = json.loads(result.stdout)
+
+        for commit in commits:
+            sha = commit['sha']
+            result = subprocess.run(
+                ['gh', 'api', f'repos/{repo_name}/contents/lean-toolchain?ref={sha}'],
+                capture_output=True,
+                text=True
+            )
+            if result.returncode == 0:
+                content = json.loads(result.stdout)
+                toolchain = base64.b64decode(content['content']).decode('utf-8').strip()
+                # Check for exact match
+                if toolchain == f'leanprover/lean4:{version}':
+                    return commit['sha']
+        return None
+    except Exception as e:
+        print(f"Error checking toolchain history for {repo['name']}: {e}", file=sys.stderr)
+        return None
+
+def get_current_toolchain(repo: Dict[str, str]) -> Optional[str]:
+    """Get the current toolchain version from the default branch"""
+    github_url = repo['github']
+    repo_name = github_url.replace('https://github.com/', '')
+
+    try:
+        result = subprocess.run(
+            ['gh', 'api', f'repos/{repo_name}/contents/lean-toolchain'],
+            capture_output=True,
+            text=True
+        )
+        if result.returncode == 0:
+            content = json.loads(result.stdout)
+            toolchain = base64.b64decode(content['content']).decode('utf-8').strip()
+            # Strip the prefix if present
+            return toolchain.replace('leanprover/lean4:', '')
+        return None
+    except Exception:
+        return None
+
+def main():
```

Comment:
Could you please let this fail gracefully if `gh` is not installed?

Reply:
Done.